### ISSUE TITLE
Investigate patching kernel bug 

### ISSUE THREAD

**Issue by moldimolt** [LumaTeam/Luma3DS#1787]
There's a bug in the 3DS kernel that, while not related to Luma3DS, causes a considerable amount of end user inconvenience and issue reports to our repository: 

```
This seems to be a kernel bug (null deref), blame Nintendo I guess.

There's a race condition in the kernel between the moments when:

* CreateThread/Run attaches the KThread to a KDebugThread
* The thread runs or any scenario involving thread->debugThread->x

Basically you have to pray this doesn't happen.
```

It may be worth investigating if it's possible to patch this bug, as it is not likely Nintendo will do so.

**Comment by PabloMK7** [comment]
Is that all the information you have related to that bug? Can you give more context on how to reproduce it? (What is your repository)

**Comment by moldimolt** [comment]
@PabloMK7 This bug is one of the most common causes of random crashes, particularly with cheats, i.e. #1272, #1779.

**Comment by OctoLinkYT** [comment]
Can confirm that it seems to happen commonly with Pokemon titles. It definitely wasn't always a thing as I had followed the exact same set of actions with no issues a few years ago.

(Just as a note, I do not condone cheating, I just use out-of-bounds cheats because I like to see the map out-of-bounds)

**Comment by Truinto** [comment]
Supposedly these crashes happen more often with some 3DS and not with others.

The main README.md acknowledges this bug.
> * **Cheat engine crashes with some applications, in particular Pokémon games**: there is a race condition in Nintendo's `Kernel11` pertaining to attaching a new `KDebugThread` to a `KThread` on thread creation, and another thread null-dereferencing `thread->debugThread`. This causes the cheat engine to crashes games that create and destroy many threads all the time (like Pokémon).
>    * For these games, having a **dedicated "game plugin"** is the only alternative until `Kernel11` is reimplemented.

It also states game plugins would work. I tested that and it does seem to work a lot better with [Gen6CTRPluginFramework](https://github.com/biometrix76/Gen6CTRPluginFramework). ~~But it can still crash randomly. I suspect not all memory modifications are equal. The code uses `MemoryManager` or `Process::Write32`. Anyone aware of plugin documentation or how these functions work? I couldn't find anything.~~ Edit: I tested it in an emulator where this kernel bug does not happen. The plugin still crashes with specific functions, so it's clearly a fault in the plugin itself. Ergo the plugin approach does actually work. Still very thin on documentation. I found a bit of comments in the header files, but not helpful.
